Fix to create all folders during webdav uploads

jdrossl opened the pull request:
https://github.com/craftercms/craftercms/issues/2581

jdrossl committed:
Fix to create all folders during webdav uploads

avasquez614 commented on src/main/java/org/craftercms/studio/impl/v1/service/webdav/WebDavServiceImpl.java:
```diff
@@ -180,22 +180,32 @@ public String upload(@ValidateStringParam(name = "site_id") final String site,
+                        sardine.createDirectory(uploadUrl);
+                        logger.debug("Folder created");
+                    } catch (Exception e) {
+                        logger.debug("Folder already exists");
```
(Same as with https://github.com/craftercms/studio/pull/1339) Wouldn't it be better to first check if the dir exists and then create it? Instead of trying to create it and then ignore the exception if it already exists. Also what if the exception is a connection issue?

jdrossl committed:
Improvements for webdav:
- Better url handling to avoid missing/duplicated slashes
- Simpler checks for folders

jdrossl committed:
Improvements for debugging
